src/clr-angular/utils/smart-popover/providers/smart-popover-events.service.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ *
+ */
+
+import { ElementRef, Injectable, Renderer2, OnDestroy, Inject } from '@angular/core';
+import { ClrSmartPopoverToggleService } from './smart-popover-toggle.service';
+import { Observable, fromEvent, Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { DOCUMENT } from '@angular/common';
+
+@Injectable()
+export class ClrSmartPopoverEventsService {
+  public outsideClickClose = true;
+  public scrollToClose = true;
+  private documentClickListener: () => void;
+  public ignoredEvent: any;
+
+  constructor(
+    private renderer: Renderer2,
+    private smartOpenService: ClrSmartPopoverToggleService,
+    @Inject(DOCUMENT) private document: HTMLDocument
+  ) {
+    smartOpenService.openChange.subscribe(change => {
+      if (change) {
```

Comment:
could we name the parameter `open` instead? Makes it a bit more clearer.

Reply:
Sure. I'll change it.